apps/serdes/Printer.h

```diff
@@ -0,0 +1,27 @@
+#ifndef HALIDE_PRINTER_H
+#define HALIDE_PRINTER_H
+
+#include <string>
+#include <vector>
+
+#include <Halide.h>
+using namespace Halide;
+
+class Printer {
+public:
+    Printer() = default;
+
+    void print_pipeline(const Halide::Pipeline &pipeline);
```

Comment:
I'm a little confused about the purpose of this class: it appears to be designed to emit human-readable text (so it's not useful for any serialization/deserialization purpose)... but we already have something like that in Halide in the `IRPrinter` class. Was it written just to help with your development effort, or does it have a use that I am overlooking?

Reply:
it's mainly for testing purposes. It does a pretty similar job as `IRPrinter` but supports more structs like `Bound` and `StorageDim`. Right now I use it to carry out roundtrip test before/after serialization as I'm still uncertain about the correct test strategy